### DISCUSSION TITLE
Custom control theme library - pop-ups not displayed in WASM build

### DISCUSSION THREAD

**Discussion by pixsperdavid** [AvaloniaUI/Avalonia#16863] [Q&A] [answered]
Currently developing a custom control theme library based on Fluent. Having an issue whereby in a WASM build of my control catalog demo application, pop-up elements are not displayed. (For context menu, combobox, etc.). In the desktop build of the same theme, I'm not having this issue, the pop-ups display correctly. I have tested with all available WASM renderers. Using Avalonia 11.1.3.

EDIT:
Found some more detail, this error is visible in the browser console when trying to open a popup:

```
Uncaught ManagedError: Unable to create IPopupImpl and no overlay layer is found for the target control
    at Jn (marshal-to-js.ts:349:18)
    at kr (invoke-cs.ts:277:19)
    at l.javaScriptExports.call_delegate (managed-exports.ts:171:13)
    at i (marshal-to-js.ts:197:53)
    at HTMLDivElement.u (input.ts:177:13)
```

**Comment by rabbitism** [comment]
You need to implement `OverlayPopupHost`

**Reply by pixsperdavid** [reply]
I've already provided a theme for `OverlayPopupHost` as part of the theme library (which is basically copy-pasted from the Fluent theme).  Is there anything else I need to do?

**Reply by rabbitism** [reply]
> I've already provided a theme for `OverlayPopupHost` as part of the theme library (which is basically copy-pasted from the Fluent theme). Is there anything else I need to do?

There shouldn't be.  You can alos set desktop popup to user overlay and check in DevTools to see if anything is missing. 

```
    public static AppBuilder BuildAvaloniaApp()
        => AppBuilder.Configure<App>()
            .UseManagedSystemDialogs()
            .UsePlatformDetect()
            .With(new Win32PlatformOptions(){ OverlayPopups = true}) <-- add this
            .LogToTrace();
```

**Reply by pixsperdavid** [reply]
Just tried that, works as expected on the desktop build.

**Comment by pixsperdavid** [comment, marked as answer]
Ok resolved, I was missing `EmbeddableControlRoot`.

As a bit of feedback, it would be much easier to develop custom themes if the  the minimum required `ControlTheme` implementations for a functional theme were documented somewhere. There's quite a few controls which are really 'internals' and not directly used in UI but if missing cause things to break.
